lib/minitest/reporters/junit_reporter.rb

```diff
@@ -26,27 +27,36 @@ def report
         puts "Writing XML reports to #{@reports_path}"
         suites = tests.group_by(&:class)
 
-        suites.each do |suite, tests|
-          suite_result = analyze_suite(tests)
-
-          xml = Builder::XmlMarkup.new(:indent => 2)
-          xml.instruct!
-          xml.testsuite(:name => suite, :skipped => suite_result[:skip_count], :failures => suite_result[:fail_count],
-                        :errors => suite_result[:error_count], :tests => suite_result[:test_count],
-                        :assertions => suite_result[:assertion_count], :time => suite_result[:time]) do
-            tests.each do |test|
-              xml.testcase(:name => test.name, :classname => suite, :assertions => test.assertions,
-                           :time => test.time) do
-                xml << xml_message_for(test) unless test.passed?
-              end
-            end
+        if @single_file
+          write_xml_file_for("minitest", suites.values.flatten)
```

Comment:
use class to sort is ok for me too.  The main reason to sort is to get stable order to simplify report comparison.